### ISSUE TITLE
TextSpan could use Tooltip support 

### ISSUE THREAD

**Issue by SunFoxx** [flutter/flutter#51587]
It would be really useful to add tooltip property to a TextSpan so you can pass a tooltip to the particular part of the text

**Comment by Chinmay-KB** [comment]
I would like to look into this, if the team plans to add this to TextSpan.

**Comment by tahirgani07** [comment]
Hi, I am new to open source and I would like to work on this.

**Comment by campovski** [comment]
Hi there, I would love to work on this too, if @tahirgani07 did not start yet.

**Comment by tahirgani07** [comment]
@campovski I have started working on it, but I am a beginner and I am facing some issues, but I will try to do it. I will inform if I will be unable to do it in a few days.

**Comment by campovski** [comment]
@tahirgani07 Acknowledged. Good luck, you can do it! 💪

**Comment by tahirgani07** [comment]
I am trying to do this but, I can't figure out a few things, if anyone would like to give me a rough idea on how to add a tooltip property for TextSpan, it would be really helpful. Thank you.🙂

**Comment by campovski** [comment]
@tahirgani07 Take a look at `IconButton`'s implementation. `IconButton` already has a `tooltip` property so adding it to `TextSpan` should be pretty much the same.

**Comment by tahirgani07** [comment]
@campovski I tried to do that but the problem is that `IconButton` extends `StatelessWidget` so you get a `build` method to return a `widget`, but `TextSpan` extends `InlineSpan` and you cannot return a `widget` and that is where i'm stuck. Any suggestions? Thank you.

**Comment by campovski** [comment]
Yes, I have an idea but currently don't have time to research it further. If we strive for simplicity, a possible solution would be to change current `InlineSpan` into `_InlineSpan` and create new `class InlineSpan extends StatelessWidget` for example wherein you create `_InlineSpan`. This was a quick thought during lunch, but the impacts would need to be checked, e.g. `Text.rich` takes `InlineSpan` as an argument. This change would affect also other descendants of `InlineSpan` (which could also use `tooltip` property IMHO, but would need to be discussed further). I will check this idea in the evening CET, if you have time to check it out I am interested in your findings.

**Comment by tahirgani07** [comment]
@campovski I'll definitely give it a try👍. Thanks for all the help. I really appreciate it.

**Comment by campovski** [comment]
I am a bit busy at my work this week so sadly I wasn't able to check it out yesterday myself. Hope it would be of any help.

**Comment by jp803333** [comment]
Is anyone still working on this ??

**Comment by tahirgani07** [comment]
@jp803333 I am currently stuck, so you can work on it.

**Comment by jp803333** [comment]
@tahirgani07 yeah same here I am also trying

**Comment by ahmedelsayed96** [comment]
Hello 
is anyone working on this ?!

**Comment by ValentinVignal** [comment]
Hello, I tried to investigate this a bit and I don't think we can make `TextSpan` a `StatelessWidget` as it was suggested.

I tried to take a look at this PR: https://github.com/flutter/flutter/issues/77754 that added the `onEnter` and `onExit` parameters to the `TextSpan` to see if was reusable but I didn't get anything. The issue is that `TextSpan` is used in `RichText` which creates a `RenderParagraph` and a `TextPainter` and I have no clue on how to get the position of the current text span and display a tooltip from those objects.

Moreover, it was a bit useless/hard to go try to go deep in the implementation of the objects as it is calling native functionalities which I'm definitely not familiar with.

I'm not sure this issue should still be labeled as `good first contribution` and `easy fix`

**Comment by AmanNegi** [comment]
Hey there I would love to work on this issue, is anyone else working on it? If not can I work on it?

**Comment by stMerlHin** [comment]
Hey there. Hope you all fine. I would like to contribute to flutter framework. Am a beginner. I notice that flutter master channel has in some source file a `switch` block without a `case` so my intellij juste underline these line and tells ""The 'case' shouldn't complete normally.". I check other branch such as `flutter-3.7-candidate.10` , everything seems ok, `case` are added to the `switch` block so I wonder if there is a particular version of dart we are supposed to work with on master branch. Thanks.

**Comment by fischerscode** [comment]
Just had a look at it...

I've considered adding a `TooltipSpan` extending `InlineSpan` and implementing `HitTestTarget`.
`TooltipSpan` should have a property `child` being an `InlineSpan`. Most of the `InlineSpan`-behavior could just be passed to the `child`. Additionally, `TooltipSpan` would have to implement the entire `TooltipState` logic.
One could probably do some clever refactoring to avoid duplicate code.

**But**, at one point, we would need an `TickerProvider` inside the `TooltipSpan` to do the animation stuff. And that seems to be a problem. A solution to this could be the addition of a `TooltipController`, that has to be provided when creating a `TooltipSpan`.

However, that's where I'll leave it for now. Feel free to implement my idea or let me know what you think about it.

**Comment by aarushlohit** [comment]
Hey there I would love to work on this issue, is anyone else working on it? If not can I work on it?

**Comment by mozammal-hossain** [comment]
As none is continuing this, I've given a PR adding tooltip on TextSpan. Please Review.

**Comment by mozammal-hossain** [comment]
@TahaTesser , I've given a PR on this issue. Could you assign me on this?
